esp8266/modules/webrepl_setup.py

```diff
@@ -1,83 +1,95 @@
+    with open(RC) as f:
+        for l in f:
+            if "webrepl" in l:
+                if l.startswith("#"):
+                    return False
+                return True
+        return None
+
+def copy_stream(s_in, s_out):
+    buf = bytearray(64)
     while 1:
-        c = stream.read(1)
-        if c in (b"\r", b"\n"):
-            stream.write("\r\n")
-            return passwd
-        passwd += c
-        stream.write("*")
-
-def handle_conn(listen_sock):
-    cl, remote_addr = listen_sock.accept()
-
-    print("""
-
-First-time WebREPL connection has been received. WebREPL initial setup
-will now start over this connection. During setup, UART REPL will be
-non-responsive. After setup finishes, the board will be rebooted. In
-case of error during setup, current session will continue.
-
-If you receive this message unexpectedly, it may mean that your WebREPL
-connection is being hacked (power off board if unsure).
-""")
-
-    websocket_helper.server_handshake(cl)
-    ws = websocket(cl)
-
-    ws.write("""\
-Welcome to MicroPython WebREPL!\r
-\r
-This is the first time you connect to WebREPL, so please set a password\r
-to use for the following WebREPL sessions. Once you enter the password\r
-twice, your board will reboot with WebREPL running in active mode. On\r
-some boards, you may need to press reset button or reconnect power.\r
-\r
-""")
-
+        sz = s_in.readinto(buf)
+        s_out.write(buf, sz)
+
+def add_daemon():
+    with open(RC) as old_f, open(RC + ".tmp", "w") as new_f:
+        new_f.write("import webrepl\nwebrepl.start()\n")
+        copy_stream(old_f, new_f)
+
+def change_daemon(action):
+    LINES = ("import webrepl", "webrepl.start()")
+    with open(RC) as old_f, open(RC + ".tmp", "w") as new_f:
+        for l in old_f:
+            for patt in LINES:
+                if patt in l:
+                    if action and l.startswith("#"):
+                        l = l[1:]
+                    elif not action and not l.startswith("#"):
+                        l = "#" + l
+            new_f.write(l)
+    os.rename(RC + ".tmp", RC)
+
+
+def getpass(x, prompt):
+    return input(prompt)
+
+def input_pass():
+    ws = None
     while 1:
         passwd1 = getpass(ws, "New password: ")
         if len(passwd1) < 4:
-            ws.write("Password too short\r\n")
+            print("Password too short")
             continue
         elif len(passwd1) > 9:
-            ws.write("Password too long\r\n")
+            print("Password too long")
             continue
         passwd2 = getpass(ws, "Confirm password: ")
         if passwd1 == passwd2:
-            break
-        ws.write("Passwords do not match\r\n")
+            return passwd1
+        print("Passwords do not match")
+
+
+status = get_status()
+
+print("webrepl daemon current status:", "enabled" if status else "disabled")
+print("\nWould you like to (E)nable or (D)isable it running on boot?")
+print("(Empty line to quit)")
+resp = input("> ").upper()
```

Comment:
Why all this code? Why not just pass that as an argument to a function -- you need to call that function anyways.